### PR TITLE
Checked files and download/mirroring

### PR DESCRIPTION
**Pull request by agarciamontoro**
TODO:

- [x] ~Exectute `release:testOnly *MirrorInS3` to mirror all releases with hash metadata.~ 
- [ ] ~Test everything with all data uploaded.~ <- I'm leaving this for the new release.
- [x] Do not execute the tests that check the files are ok for the non-mirrored versions ~(which should be mirrored as a consquence)~.
- [x] Wait for the new release of s3 with `getCheckedFileIfDifferent` and update the `s3` dependency here.

### CODE REVIEW

**Comment by agarciamontoro** [comment]
This PR ended up implementing features slightly different from the ones originally intended, basically following the ideas in [this comment](https://github.com/ohnosequences/16s/issues/67#issuecomment-430200892)

**Comment by agarciamontoro** [comment]
After agreeing on [a new release process](https://github.com/ohnosequences/16s/issues/67#issuecomment-430200892) for repositories that generate data resources, this PR changed its original purpose and ended up addressing slightly different issues.

* The release process was refactored out of tests and reimplemented in an object on the main code.
* The definition of the data was slightly improved
* The old versions were removed from the code and the current test versions are being uploaded to an `unstable` subdirectory in S3.
* A new test was implemented to check that all supported versions are in S3 as expected.

These changes on the code were tested and version 10.0 of RNACentral is already on S3, under 

```
s3://resources.ohnosequences.com/ohnosequences/db/rnacentral/unstable/10.0/
```

I'll wait until [the fix for changeMetadata](https://github.com/ohnosequences/s3/pull/32) is completely finished and tested so I can ask for a review as soon as possible.

**Comment by agarciamontoro** [comment]
After agreeing on [a new release process](https://github.com/ohnosequences/16s/issues/67#issuecomment-430200892) for repositories that generate data resources, this PR changed its original purpose and ended up addressing slightly different issues.

* The release process was refactored out of tests and reimplemented in an object on the main code.
* The definition of the data was slightly improved
* The old versions were removed from the code and the current test versions are being uploaded to an `unstable` subdirectory in S3.
* A new test was implemented to check that all supported versions are in S3 as expected.

These changes on the code were tested and version 10.0 of RNACentral is already on S3, under 

```
s3://resources.ohnosequences.com/ohnosequences/db/rnacentral/unstable/10.0/
```

I'll wait until [the fix for changeMetadata](https://github.com/ohnosequences/s3/pull/32) is completely finished and tested so I can ask for a review as soon as possible.

**Comment by agarciamontoro** [comment]
After [v0.2.1 is released](https://github.com/ohnosequences/s3/pull/33), we can safely finish this.

**Comment by agarciamontoro** [comment]
Friendly ping to @ncordon :see_no_evil:

**Comment by ncordon** [comment]
Sorry for the delay. Doing this now!

File: src/test/scala/data.scala [hunk 1]
```diff
@@ -4,6 +4,8 @@ import ohnosequences.db.rnacentral
 import ohnosequences.fastarious.fasta._
 import rnacentral.{EntryAnnotation, RNACentralData, RNAID, Version, iterators}
 import java.io.File
+import ohnosequences.s3.{Error => S3Error}
+import rnacentral.s3Helpers.getCheckedFileIfDifferent
 
 object data {
```
Comment (ncordon) [inline src/test/scala/data.scala]: Just a suggestion: maybe `cleanLocalFolder` could be erased from here (it is not being used anywhere else). Apart from that we know have an specific function in `files` to delete directories that outputs `File.Error` if something went wrong.

Comment (agarciamontoro) [reply src/test/scala/data.scala]: Oops, legacy code I forgot to delete, thanks!

**Pull request merged**